weblate/trans/models/component.py

```diff
@@ -1179,6 +1211,95 @@ def get_repoweb_link(
+
+        if (
+            self.repo.startswith("git@github.com")
+            or parsed_url.hostname == "github.com"
+        ):
+            return self.get_github_repoweb_template()
+
+        if "pagure.io" in self.repo:
+            return self.get_pagure_repoweb_template()
+
+        if (
+            self.repo.startswith("git@ssh.dev.azure.com:v3")
+            or parsed_url.hostname == "dev.azure.com"
+            or "visualstudio.com" in self.repo
+        ):
+            return self.get_azure_repoweb_template()
+
+        return None
+
+    def get_bitbucket_git_repoweb_template(self):
+        owner, slug, matches = None, None, None
+        if re.match(BITBUCKET_GIT_REPOS_REGEXP[0], self.repo):
+            matches = re.search(BITBUCKET_GIT_REPOS_REGEXP[0], self.repo)
+        elif re.match(BITBUCKET_GIT_REPOS_REGEXP[1], self.repo):
+            matches = re.search(BITBUCKET_GIT_REPOS_REGEXP[1], self.repo)
+        if matches:
+            owner = matches.group(1)
+            slug = matches.group(2)
+        if owner and slug:
+            return f"https://bitbucket.org/{owner}/{slug}"
+
+        return None
+
+    def get_github_repoweb_template(self):
+        owner, slug, matches = None, None, None
+        if re.match(GITHUB_REPOS_REGEXP[0], self.repo):
+            matches = re.search(GITHUB_REPOS_REGEXP[0], self.repo)
+        elif re.match(GITHUB_REPOS_REGEXP[1], self.repo):
+            matches = re.search(GITHUB_REPOS_REGEXP[1], self.repo)
+        if matches:
+            owner = matches.group(1)
+            slug = matches.group(2)
+        if owner and slug:
+            return f"https://github.com/{owner}/{slug}"
+
+        return None
+
+    def get_pagure_repoweb_template(self):
+        owner, slug = None, None
+        if re.match(PAGURE_REPOS_REGEXP[0], self.repo):
+            matches = re.search(PAGURE_REPOS_REGEXP[0], self.repo)
+            owner = matches.group(1)
+            slug = matches.group(2)
+
+        if owner and slug:
+            return f"https://pagure.io/{owner}/{slug}"
+
+        return None
+
+    def get_azure_repoweb_template(self):
+        organization, project, repository, matches = None, None, None, None
+        if re.match(AZURE_REPOS_REGEXP[0], self.repo):
+            matches = re.search(AZURE_REPOS_REGEXP[0], self.repo)
+        elif re.match(AZURE_REPOS_REGEXP[1], self.repo):
+            matches = re.search(AZURE_REPOS_REGEXP[1], self.repo)
+        elif re.match(AZURE_REPOS_REGEXP[2], self.repo):
+            matches = re.search(AZURE_REPOS_REGEXP[2], self.repo)
+        elif re.match(AZURE_REPOS_REGEXP[3], self.repo):
+            matches = re.search(AZURE_REPOS_REGEXP[3], self.repo)
+
+        if matches:
+            organization = matches.group(1)
+            project = matches.group(2)
+            repository = matches.group(3)
+
+        if organization and project and repository:
+            return f"https://dev.azure.com/{organization}/{project}/_git/{repository}"
```

Comment:
Again, this needs to be a template to generate the URL for displaying the file in the web browser, see above.